R-packages/modeltools/tests/testthat/test-matrix.R

```diff
@@ -0,0 +1,44 @@
+library(tibble)
+
+test_that("training and prediction matrices are created", {
+    df <- tibble(
+        geo_value = rep(c("az", "wv"), 5),
+        time_value = rep(
+            as.Date(c("2021-01-25", "2021-01-26", "2021-01-27", "2021-01-28", "2021-01-29")),
+            each = 2),
+        `value-2:signal_1` = seq(-3, 6),
+        `value-2:signal_2` = seq(7, 16),
+        `value-1:signal_1` = seq(-1, 8),
+        `value-1:signal_2` = seq(9, 18),
+        `value+0:signal_1` = seq(1, 10),
+        `value+0:signal_2` = seq(11, 20),
+        `response+1:signal_1` = c(seq(3, 10), rep(NA, 2)),
+        `response+1:signal_2` = c(seq(13, 20), rep(NA, 2)),
+        `response+2:signal_1` = c(seq(5, 10), rep(NA, 4)),
+        `response+2:signal_2` = c(seq(15, 20), rep(NA, 4))
+    )
+
+    out <- create_train_and_predict_matrices(df, 2, 1)
+
+    expect_equal(names(out), c("train_x", "train_y", "predict_x", "predict_geo_values"))
+    expect_equal(out$train_x,
+                 as.matrix(tibble(
+                    `value-2:signal_1` = c(1, 2),
+                    `value-2:signal_2` = c(11, 12),
+                    `value-1:signal_1` = c(3, 4),
+                    `value-1:signal_2` = c(13, 14),
+                    `value+0:signal_1` = c(5, 6),
+                    `value+0:signal_2` = c(15, 16)))
+    )
+    expect_equal(out$train_y, c(19, 20))
```

Comment:
I think this is not correct. You have two response+2 columns, so based on your code I think both of these columns should appear in `out$train_y`.

Reply:
Yes something is not right here.  Looking into it

Reply:
It turns out that `pull()` by default selects the last column and silently ignores the others.  I have a check fail in here if there are multiple response columns for the same ahead.  We could also change this to allow users to specify the response variable but I don't know if that's actually something that people would want to do in the same matrix or if they'd rather have one copy of the matrix per response (would be straightforward for users to do this with `add_reponse_columns()`.